src/java.desktop/windows/native/libawt/windows/awt_Canvas.cpp

```diff
@@ -202,23 +202,34 @@ MsgRouting AwtCanvas::HandleEvent(MSG *msg, BOOL synthetic)
     return AwtComponent::HandleEvent(msg, synthetic);
 }
 
-void AwtCanvas::_SetEraseBackground(void *param)
-{
-    JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
+void AwtCanvas::_SetEraseBackground(void *param) {
+    JNIEnv *env = (JNIEnv *) JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
-    SetEraseBackgroundStruct *sebs = (SetEraseBackgroundStruct *)param;
+    SetEraseBackgroundStruct *sebs = static_cast<SetEraseBackgroundStruct *>(param);
     jobject canvas = sebs->canvas;
     jboolean doErase = sebs->doErase;
     jboolean doEraseOnResize = sebs->doEraseOnResize;
 
     PDATA pData;
-    JNI_CHECK_PEER_GOTO(canvas, ret);
+    if (canvas == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "peer");
+        env->DeleteGlobalRef(canvas);
```

Comment:
you didn't code this up the way I did in my comment, which I think you should have done, and here you are deleting a ref to NULL, and the string passed to throw doesn't mention the canvas, which it can do unlike the old code.

Reply:
Sorry, I was trying to keep as close to the original code as possible. Woah, does that mean the original was bugged, since it was deleting a NULL canvas? Anyway, should I change the string to canvas or just keep it as "peer" like the original code does in that case?

Reply:
Not buggy per se, since NULL is allowed
https://docs.oracle.com/en/java/javase/21/docs/specs/jni/functions.html#deleteglobalref

But that doesn't make it good practice.

Yes, change the name because then it is more precise.

Reply:
Is the new code better?